test/runtests.jl

```diff
@@ -17,97 +17,156 @@ expit(X) = 1 ./ (1 .+ exp.(-X))
 ###
 ### OLSREGRESSOR
 ###
-
-X, y = @load_boston
-
-train, test = partition(eachindex(y), 0.7)
-
-atom_ols = LinearRegressor()
-
-Xtrain = selectrows(X, train)
-ytrain = selectrows(y, train)
-Xtest  = selectrows(X, test)
-
-fitresult, _, _ = fit(atom_ols, 1, Xtrain, ytrain)
-θ = MLJBase.fitted_params(atom_ols, fitresult)
-
-p = predict_mean(atom_ols, fitresult, Xtest)
-
-# hand made regression to compare
-
-Xa = MLJBase.matrix(X) # convert(Matrix{Float64}, X)
-Xa1 = hcat(Xa, ones(size(Xa, 1)))
-coefs = Xa1[train, :] \ y[train]
-p2 = Xa1[test, :] * coefs
-
-@test p ≈ p2
-
-model = atom_ols
-@test name(model) == "LinearRegressor"
-@test package_name(model) == "GLM"
-@test is_pure_julia(model)
-@test is_supervised(model)
-@test package_license(model) == "MIT"
-@test prediction_type(model) == :probabilistic
-@test hyperparameters(model) == (:fit_intercept, :dropcollinear, :offsetcol, :report_keys)
-hyp_types = hyperparameter_types(model)
-@test hyp_types[1] == "Bool"
-@test hyp_types[2] == "Bool"
-@test hyp_types[3] == "Union{Nothing, Symbol}"
-@test hyp_types[4] == "Union{Nothing, AbstractVector{Symbol}}"
-
-p_distr = predict(atom_ols, fitresult, selectrows(X, test))
-
-dispersion =  MLJGLMInterface.dispersion(fitresult)
-@test p_distr[1] == Distributions.Normal(p[1], dispersion)
+@testset "OLSREGRESSOR" begin
+    X, y = @load_boston
+    w = ones(eltype(y), length(y))
+
+    train, test = partition(eachindex(y), 0.7)
+
+    atom_ols = LinearRegressor()
+
+    Xtrain = selectrows(X, train)
+    ytrain = selectrows(y, train)
+    wtrain = selectrows(w, train)
+    Xtest  = selectrows(X, test)
+
+    # Without weights `w`
+    fitresult, _, _ = fit(atom_ols, 1, Xtrain, ytrain)
+    θ = MLJBase.fitted_params(atom_ols, fitresult)
+
+    p = predict_mean(atom_ols, fitresult, Xtest)
+
+    # With weights `w`
+    fitresultw, _, _ = fit(atom_ols, 1, Xtrain, ytrain, wtrain)
+    θw = MLJBase.fitted_params(atom_ols, fitresult)
+
+    pw = predict_mean(atom_ols, fitresultw, Xtest)
+
+    # hand made regression to compare
```

Comment:
I'd say we can omit these comments. Readers are double checking the comment to the code anyway so they don't add much. In general, answer "why" in comments and not "what". "what" is already answered by the code in great detail.